wgsl/index.bs

```diff
@@ -7103,32 +7103,73 @@ The second phase explores that graph, resulting in either rejecting the program,
+  * A <dfn noexport>function tag</dfn> describing the function's effects on uniformity.
 
-* A tag about the control-flow in which the function can be called: one of {AlwaysRequiredToBeUniform, RequiredToBeUniformForControlFlowAfter}
-* A tag for each parameter of the function: one of {AlwaysRequiredToBeUniform, RequiredToBeUniformForControlFlowAfter, RequiredToBeUniformForReturnValue, NoRestriction}
-* A tag for the function as a whole: one of {ControlFlowMayBeNonUniformAfter, ReturnValueMayBeNonUniform, NoRestriction}
+Additionally, for each [=formal parameter=] of a function, a <dfn
+noexport>parameter tag</dfn> is computed that describes the uniformity
+requirement of the parameter value.
 
-Here is the algorithm for computing this data for a given function:
+<table class='data'>
+  <caption>[=Call site tag=] values</caption>
+  <thead>
+    <tr><th>Call Site Tag<th>Description
+  </thead>
+  <tr><td><dfn noexport>CallSiteRequiredToBeUniform</dfn>
+      <td>The function must only be called from [=uniform control flow=].
+  <tr><td><dfn noexport>CallSiteNoRestriction</dfn>
+      <td>The function may be called from [=uniform control flow|non-uniform control flow=].
+</table>
+
+<table class='data'>
+  <caption>[=Function tag=] values</caption>
+  <thead>
+    <tr><th>Function Tag<th>Description
+  </thead>
+  <tr><td><dfn noexport>SubsequentControlFlowMayBeNonUniform</dfn>
+      <td>The control flow after the function call may be non-uniform.
+  <tr><td><dfn noexport>ReturnValueMayBeNonUniform</dfn>
+      <td>The [=return value=] of the function may be non-uniform.
+  <tr><td><dfn noexport>NoRestriction</dfn>
+      <td>The function does not introduce non-uniformity.
+</table>
+
+<table class='data'>
+  <caption>[=Parameter tag=] values</caption>
+  <thead>
+    <tr><th>Parameter Tag<th>Description
+  </thead>
+  <tr><td><dfn noexport>ParameterRequiredToBeUniform</dfn>
+      <td>The parameter must be a [=uniform value=].
+  <tr><td><dfn noexport>ParameterRequiredToBeUniformForSubsequentControlFlow</dfn>
+      <td>The parameter must be a [=uniform value=] for control flow after the function call to be [=uniform control flow|uniform=].
+  <tr><td><dfn noexport>ParameterRequiredToBeUniformForReturnValue</dfn>
+      <td>The parameter must be a [=uniform value=] in order for the [=return value=] to be a uniform value.
+  <tr><td><dfn noexport>ParameterNoRestriction</dfn>
+      <td>The parameter value has no uniformity requirement.
+</table>
+
+The following algorithm describes how to compute these tags for a given function:
 
-* Create nodes called "RequiredToBeUniform", "MayBeNonUniform", "CF_start", "CF_return", and if the function is non-void a node called "Value_return"
-* Create one node for each parameter of the function which we'll call "arg_i"
+* Create nodes called "RequiredToBeUniform", "MayBeNonUniform", "CF_start", "CF_return", and if the function is non-void a node called "Value_return".
```

Comment:
instead of "the function is non-void".... say "if the funciton has a [=return type=]".